base/system/winlogon/winlogon.h

```diff
@@ -285,6 +286,24 @@ extern PWLSESSION WLSession;
    ((Status) == WLX_SAS_ACTION_SHUTDOWN_HIBERNATE) \
   )
 
+FORCEINLINE
+VOID
+SetLogonTimestamp(
+    _Inout_ PWLSESSION Session)
+{
+    GetSystemTime(&Session->LastLogon);
```

Comment:
Or maybe prefer `NtQuerySystemTime(&Session->LastLogon);` with a `LARGE_INTEGER LastLogon;` , since there doesn't seem to be a need to have a structured time (with definite day, hour, etc. fields).
This would also avoid the comment in the other function below, you would just do `return (Session->LastLogon.QuadPart == 0);` and call it a day.

Reply:
OK.

Reply:
I think `SYSTEMTIME` is more useful if we (maybe) want to show this time to the user (in the future).